Content.Server/Tesla/Components/TeslaEnergyBallComponent.cs

```diff
@@ -0,0 +1,58 @@
+using Robust.Shared.Audio;
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Tesla.Components;
+
+/// <summary>
+/// A component that tracks an entity's saturation level from absorbing other creatures by touch, and spawns new entities when the saturation limit is reached.
+/// </summary>
+[RegisterComponent]
+public sealed partial class TeslaEnergyBallComponent : Component
+{
+
+    public float AccumulatedFrametime = 0.0f;
```

Comment:
why are you doing this instead of NextUpdate or whatever timespan

Reply:
I don't know any other way. Saw this method in the circulatory system a long time ago, memorized it, and use it everywhere. Examples of the new version?

Reply:
You used it in a few other spots so please re-use it and make sure to handle paused time correctly.